commands/cmdsets/tests.py

```diff
@@ -607,3 +607,24 @@ def test_cmd_haggle(self, mock_dice_check):
                                                 category='Deal Offer')
         self.call_cmd("/findseller testium,testaccount2=50,25",
                       "They already have a deal in progress. Ask them to decline it first.")
+
+class TestHomeCommands(ArxCommandTest):
+    def test_cmd_shop(self):
+        from world.dominion.models import CraftingRecipe
+        recipes = {CraftingRecipe.objects.create(id=1, name="Item1"),  
+                   CraftingRecipe.objects.create(id=2, name="Item2")}
+        self.char.player_ob.Dominion.assets.recipes.set(recipes)
+        prices = self.room.db.crafting_prices or {}
+        prices[1] = 10
+        prices["removed"] = {2}
+        self.room.db.crafting_prices = prices
+        self.room.db.shopowner = self.char
+        self.char.location = self.room
+        self.setup_cmd(home.CmdBuyFromShop, self.char)
+        self.call_cmd("/test", "Invalid switch.")
+        self.call_cmd("/craft Item2", "Recipe by the name Item2 is not available.")
+        self.call_cmd("/craft Item3", "No recipe found by the name Item3.")
+        self.call_cmd("", 'Crafting Prices\n\nName  Craft Price Refine Price \n'
+                          'Item1 0.0         0            \nItem Prices')
+        self.call_cmd("/craft Item1", 'You have started to craft: Item1.|'
+                                      'To finish it, use /finish after you gather the following:|Silver: 10')                 
```

Comment:
Test failed here because of the silver cost not being set in the recipe - you can change the first recipe to either have a kwarg of a silver cost of 10, or change the expected output to be `Silver: 0` here

Reply:
Yup, just fixed it.